src/plugins/data/public/ui/filter_bar/filter_bar.tsx

```diff
@@ -279,15 +284,22 @@ const FilterBarUI = React.memo(function FilterBarUI(props: Props) {
   // }
 
   const classes = classNames('globalFilterBar', props.className);
+  const { savedQueryService, onFilterSave } = props;
 
   return (
-    <EuiFlexGroup
-      className="globalFilterGroup"
-      gutterSize="none"
-      alignItems="flexStart"
-      responsive={false}
+    <FilterBarContext.Provider
+      value={{
+        savedQueryService,
```

Comment:
Nodir could you please verify that, not sure that it should be saved in context. Maybe props?